repos/system_upgrade/common/actors/checkpersistentmounts/libraries/checkpersistentmounts.py

```diff
@@ -0,0 +1,56 @@
+from leapp import reporting
+from leapp.libraries.stdlib import api
+from leapp.models import StorageInfo
+
+
+def inhibit_upgrade_due_non_persistent_mount(mountpoint):
+    summary = (
+        'Leapp detected that the {0} mountpoint is mounted, with no corresponding entry in /etc/fstab. '
+        'The upgrade process cannot continue with {0} mounted in non-persistent fashion, '
+        'as Leapp needs this mount to be available after a reboot.'
+    )
+
+    hint = (
+        'Add {0} mount entry to /etc/fstab'
+    )
+
+    reporting.create_report([
+        reporting.Title(
+            'Detected partitions mounted in a non-persistent fashion, preventing a successful in-place upgrade.'
+        ),
+        reporting.Summary(summary.format(mountpoint)),
+        reporting.Remediation(hint=hint.format(mountpoint)),
+        reporting.RelatedResource('file', '/etc/fstab'),
+        reporting.Severity(reporting.Severity.HIGH),
+        reporting.Tags([reporting.Tags.FILESYSTEM]),
+        reporting.Flags([reporting.Flags.INHIBITOR]),
+    ])
+
+
+def mount_entry_with_mountpoint_exists(mount_entries, mountpoint):
+    for mount_entry in mount_entries:
+        if mount_entry.mount == mountpoint:
+            return True
+    return False
+
+
+def fstab_entry_with_mountpoint_exists(fstab_entries, mountpoint):
+    for fstab_entry in fstab_entries:
+        if fstab_entry.fs_file == mountpoint:
+            return True
+    return False
+
+
+def check_mount_is_persistent(storage_info, mountpoint):
+    """Check if mountpoint is mounted in persistent fashion"""
+
+    mount_entry_exists = mount_entry_with_mountpoint_exists(storage_info.mount, mountpoint)
+    fstab_entry_exists = fstab_entry_with_mountpoint_exists(storage_info.fstab, mountpoint)
+
+    if mount_entry_exists and not fstab_entry_exists:
+        inhibit_upgrade_due_non_persistent_mount(mountpoint)
+
+
+def check_persistent_mounts():
+    for storage_info in api.consume(StorageInfo):
+        check_mount_is_persistent(storage_info, '/var/lib/leapp')
```

Comment:
We should check this only *once* and only for the first message received. The expectation is that there's only one StorageInfo message. Also we want to avoid multiple inhibiting reports for potentially multiple StorageInfo messages.
```suggestion
     storage_info = next(api.consume(StorageInfo), None)
     if storage_info:
        check_mount_is_persistent(storage_info, '/var/lib/leapp')
```